backend/fs/DataArrayFS.cpp

```diff
@@ -12,30 +12,41 @@
 #include "hdf5/h5x/H5DataSet.hpp" // FIXME
 #include "DimensionFS.hpp"
 
-namespace bfs = boost::filesystem;
-
 namespace nix {
 namespace file {
 
-
 DataArrayFS::DataArrayFS(const std::shared_ptr<base::IFile> &file, const std::shared_ptr<base::IBlock> &block,
-                         const std::string &loc)
+                         const bfs::path &loc)
     : EntityWithSourcesFS(file, block, loc),
-      dimensions(loc + bfs::path::preferred_separator + "dimensions", file->fileMode()) {
+      dimensions(bfs::path(loc).append("/dimensions"), file->fileMode()) {
```

Comment:
boost::filesystem automatically changes any path separators into the native format.